### PR TITLE
Notes about 1.0.0-beta.30

### PR DESCRIPTION
**Pull request by JessicaSachs**
* Adding notes about async vs. sync mode.
* Adding placeholder for mocking transitions section.

### CODE REVIEW

**Review by dobromir-hristov — APPROVED** [review]

**Comment by hiendv** [comment]
I think we could also document `wrapper.vm.$nextTick()` as an interchangeable option to `Vue.nextTick()`.

**Comment by lmiller1990** [comment]
@JessicaSachs a bit about mocking transitions, 99% of the work was done by @hiendv , I hope you don't mind I used your example and sneaky `transitionStub` helper.

I'm excited to learn how Vue's `<transition>` works under the hood (have not used it much), and how we can let it work correctly in VTU as well.

**Comment by hiendv** [comment]
@lmiller1990 It would be my honor. I've been using VTU for years but never had a chance (or time :rofl: ) to contribute back to the library. Glad you took my snippet which, to be honest, is kinda a skimmed version of [the original](https://github.com/vuejs/vue-test-utils/blob/v1.0.0-beta.29/packages/test-utils/src/components/TransitionStub.js) by @eddyerburgh. I think [the old component stubs](https://github.com/vuejs/vue-test-utils/tree/v1.0.0-beta.29/packages/test-utils/src/components) could be a good start for you.

File: docs/guides/common-tips.md [hunk 1]
```diff
@@ -160,6 +174,104 @@ You can also update the props of an already-mounted component with the `wrapper.
 
 _For a full list of options, please see the [mount options section](../api/options.md) of the docs._
 
+### Mocking Transitions
+
+Although calling `await Vue.nexTick()` works well for most use cases, there are some situations where additional work arounds are required. These issues will be solved before the library transitions out of beta. One such example is unit testing components with the `<transition>` wrapper provided by Vue.
```
Comment (FloEdelmann) [inline docs/guides/common-tips.md]: Typo:

```suggestion
Although calling `await Vue.nextTick()` works well for most use cases, there are some situations where additional workarounds are required. These issues will be solved before the library transitions out of beta. One such example is unit testing components with the `<transition>` wrapper provided by Vue.
```

I'd also clarify which library you are talking about and avoid the word *transition* (at least in this paragraph):

```suggestion
Although calling `await Vue.nextTick()` works well for most use cases, there are some situations where additional workarounds are required. These issues will be solved before the `vue-test-utils` library moves out of beta. One such example is unit testing components with the `<transition>` wrapper provided by Vue.
```

File: docs/guides/common-tips.md [hunk 2]
```diff
@@ -160,6 +174,104 @@ You can also update the props of an already-mounted component with the `wrapper.
...
+  wrapper.setData({
+    show: false
+  })
+  await wrapper.vm.$nextTick()
+
+  expect(wrapper.text()).not.toMatch(/Foo/)
+})
+```
+
+In practice, although we are calling `setData` then waiting for the `nextTick` to ensure the DOM is updated, this test fails. This is an ongoing issue related to show Vue implements the `<transition`> component, that we would like to solve before version 1.0. For now, there are some work arounds:
```
Comment (FloEdelmann) [inline docs/guides/common-tips.md]: Typo:

```suggestion
In practice, although we are calling `setData` then waiting for the `nextTick` to ensure the DOM is updated, this test fails. This is an ongoing issue related to show Vue implements the `<transition>` component, that we would like to solve before version 1.0. For now, there are some workarounds:
```

File: docs/guides/common-tips.md [hunk 3]
```diff
@@ -160,6 +174,104 @@ You can also update the props of an already-mounted component with the `wrapper.
...
+
+1. Using a `transitionStub` helper
```
Comment (FloEdelmann) [inline docs/guides/common-tips.md]: To structure this section and clearly show what belongs to it, I'd make this a heading.

```suggestion
#### Using a `transitionStub` helper
```

File: docs/guides/common-tips.md [hunk 4]
```diff
@@ -160,6 +174,104 @@ You can also update the props of an already-mounted component with the `wrapper.
...
+  })
+  await wrapper.vm.$nextTick()
+
+  expect(wrapper.text()).not.toMatch(/Foo/)
+})
+```
+
+This overrides the default behavior of the `<transition>` component and renders the children as soon as the relevant boolean condition changes, as opposed to applying CSS classes, which is how Vue's `<transition>` component works.
+
+Another alternative is to simply avoid using `setData` by writing two tests, with the required setup performed using `mount` or `shallowMount` options:
```
Comment (FloEdelmann) [inline docs/guides/common-tips.md:249]: When there's a `1.` above, here should be a `2.`, or another heading

```suggestion
#### Avoid `setData`

Another alternative is to simply avoid using `setData` by writing two tests, with the required setup performed using `mount` or `shallowMount` options:
```

```suggestion
2. Avoid `setData`

Another alternative is to simply avoid using `setData` by writing two tests, with the required setup performed using `mount` or `shallowMount` options:
```

**Review by FloEdelmann — CHANGES_REQUESTED** [review]
Coming from #1367, I've got some remarks.

File: docs/guides/common-tips.md (hunk 1 above)
Comment (JessicaSachs) [reply docs/guides/common-tips.md]: TYVM

**Comment by JessicaSachs** [comment]
These docs will help provide guidance for [1.0.0-beta.30 migration-related issues](https://github.com/vuejs/vue-test-utils/issues/1363)

**Comment by JessicaSachs** [comment]
These docs will help provide guidance for [1.0.0-beta.30 migration-related issues](#1363)

**Pull request merged**